swap_meet/vendor.py

```diff
@@ -1,2 +1,178 @@
 class Vendor:
-    pass
\ No newline at end of file
+    """
+    A class to represent a vendor.
+
+    Attributes
+    ----------
+    inventory : list
+        list of items in inventory
+        optional attribute defaulted aas None 
+        Uses logic to create empty dictionary if needed
+
+    Methods
+    -------
+    add(item):
+        appends new item to inventory list
+    remove(item):
+        removes item from inventory list
+        if item not in list, raises ValueError and returns False
+    """
+
+    def __init__(self, inventory = None):
+        """
+        Constructs all the necessary attributes for the vender object.
+        """
+        if not inventory:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        """Takes in an item and appends to inventory list."""
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        """Removes specific item from inventory list if available, if not, returns False."""
+        try:
+            self.inventory.remove(item)
+            return item
+        except ValueError:
+            return False
+
+    def get_by_category(self, category):
+        """
+        Returns list of items with a given category. 
+
+        If there are no items with the given category, it returns an empty list.
+
+        Parameters
+        ----------
+        category : str
+            string of the category each item must match to be added to the list
+
+        Returns
+        -------
+        category items list
+        """
+        category_items = []
+        # try:
+        for item in self.inventory:
+            if item.category == category:
+                category_items.append(item)
+        return category_items
+        # except ValueError:
+            # pass
+            # return category_items
+
+    def swap_items(self, other_vender, my_item, their_item):
+        """
+        Returns True if items have been swapped between two inventory lists of two objects. 
+
+        If the vendor's inventory or other vender's invenory is an empty list, returns False.
+        If either of the items to be swapped not in appropriate lists, returns False
+
+        Parameters
+        ----------
+        other_vender: Vender instance object
+            The instance of the class Vendor
+        my_item: component
+            Item instance with given cateogry to be removed from self.inventory and appended to other_vender
+        their_item: component
+            Item instance with given cateogry to be removed from other_vender.inventory and appended to self.inventory
+
+        Returns
+        -------
+        True/False
+        """
+        if my_item not in self.inventory or their_item not in other_vender.inventory:
+            return False
+        elif not self.inventory or not other_vender.inventory:
+            return False
```

Comment:
Hmm, not sure this second guard clause is necessary. But if we keep both, I'd suggest swapping the conditional statements, so it checks if the `inventory` is empty first. It's a very small detail that comes down to personal preference, so don't put too much weight on this suggestion.